java/gandiva/src/main/java/org/apache/arrow/gandiva/evaluator/JniLoader.java

```diff
@@ -35,7 +37,8 @@
 
   private static volatile JniLoader INSTANCE;
   private static volatile long defaultConfiguration = 0L;
-  private static volatile long unoptimizedConfiguration = 0L;
+  private static final ConcurrentMap<ConfigurationBuilder.ConfigOptions, Long> configurationMap
```

Comment:
Do we have tests to ensure that this map works as expected?

Using the same Config should return the same long. Release should work as expected, etc.

Reply:
I see a releaseConfig in JNI. Dont we need something similar here to reduce the size of the Map?